### PR TITLE
Recognize softwares as mission item

### LINKED ISSUE #42552: Items within SOFTWARE type pockets cant be turned in for missions
Describe the bug
Missions that ask for items dont recognize items stored within software type pockets. Making the USB npc mission and the `Steal a Dead Man's Mind` missions impossible to complete

Steps To Reproduce
Spawn an NPC and give them the 'Retrieve Software' mission. Receive an empty usb.
Travel to the mission target computer and chose the download software computer option
Return carrying the  Usb with software, and notice you can give the item to the npc 

Expected behavior
The software item can be turned in to complete the mission

### PR DESCRIPTION
**Pull request by Menschheit**
#### Summary

SUMMARY: Bugfixes "Make if possible to complete the two missions associated with software items"

#### Purpose of change

fix #42552

#### Describe the solution

Made some adjustments to manually include softwares for the two missions.

#### Describe alternatives you've considered

Add an extra parameter for all methods in the calling chain.

#### Testing

Completed "retrieve software" normally.

#### Additional context

Second try.

May still be a bad idea.

### CODE REVIEW

File: src/mission.cpp [hunk 1]
```diff
@@ -389,14 +390,21 @@ bool mission::is_complete( const character_id &_npc_id ) const
...
+                search_modifier::mode = 0b1;
+                return ret;
             }
+            search_modifier::mode = 0b1;
         }
         return true;
 
         case MGOAL_FIND_ANY_ITEM:
-            return player_character.has_mission_item( uid ) && ( !npc_id.is_valid() || npc_id == _npc_id );
+            search_modifier::mode = 0b100001;
```
Comment (Qrox) [inline src/mission.cpp:404]: I would not recommend using a magic number and global variable for the search mode.

File: src/visitable.cpp [hunk 2]
```diff
@@ -390,7 +390,7 @@ VisitResponse item_contents::visit_contents( const std::function<VisitResponse(
         &func, item *parent )
 {
     for( item_pocket &pocket : contents ) {
-        if( !pocket.is_type( item_pocket::pocket_type::CONTAINER ) ) {
+        if( !( ( 0b1 << pocket.get_pocket_data()->type ) & search_modifier::mode ) ) {
             // anything that is not CONTAINER is accessible only via its specific accessor
```
Comment (Qrox) [inline src/visitable.cpp:394]: How about adding a new accessor for the `SOFTWARE` pocket type?

**Comment by jbytheway** [comment]
What if the visitor just visited all `SOFTWARE` pockets by default?  @KorGgenT is there a particular reason it's limited to `CONTAINER` pockets?

**Comment by KorGgenT** [comment]
Because CONTAINER is the only pocket type that should be able to be interacted with normally. every other pocket type needs some kind of special accessor, like MAGAZINE and MAGAZINE_WELL have reloading and unloading code, and MOD has the add and remove mod code.

**Comment by KorGgenT** [comment]
as this is one of the pieces of code that is the absolute most arcane, i don't think this solution is going in the right direction, and "temporary" isn't likely going to be as temporary as you think. I agree with Qrox that there should just be an accessor, and then look through those accessors in visit_items to find the software item.

**Comment by Menschheit** [comment]
Second try.
Still, it worked, but I dont know if I'm in the right direction.

**Comment by KorGgenT** [comment]
yeah this looks far better imo, good work.

**Pull request merged**